### DISCUSSION TITLE
How to change button style when pointer is over or button is pressed?

### DISCUSSION THREAD

**Discussion by wnsntgs** [AvaloniaUI/Avalonia#7955] [Q&A] [answered]
* My code:
```
<Grid.Styles>
    <Style Selector="Button:pointerover">
        <Setter Property="Foreground" Value="Red" />
    </Style>
    <Style Selector="Button:pressed">
        <Setter Property="Background" Value="Purple" />
    </Style>
    <Style Selector="Button">
        <Setter Property="Background" Value="Transparent" />
        <Setter Property="Foreground" Value="Blue" />
    </Style>
</Grid.Styles>
```
Buttons inside the grid have transparent background and blue foreground, but `pressed` and `pointerover` styles don't work.

* Code from documentation:
```
<!-- Here #PART_ContentPresenter name selector is not necessary, but was added to have more specific style -->
<Style Selector="Button:pointerover /template/ ContentPresenter#PART_ContentPresenter">
    <Setter Property="Background" Value="Blue" />
</Style>
``` 
I can't build the project `Unable to resolve suitable regular or attached property Foreground on type Avalonia.Controls:Avalonia.Controls.Presenters.ContentPresenter`

Is there any good documentation for working with styles?

**Comment by timunie** [comment]
Tbh I always look this up in the source (original styles) and copy the needed parts over into my App.

Happy coding
Tim

**Comment by amwx** [comment, marked as answer]
> I can't build the project `Unable to resolve suitable regular or attached property Foreground on type Avalonia.Controls:Avalonia.Controls.Presenters.ContentPresenter`

Use `TextBlock.Foreground` instead, most of the text properties and Foreground don't exist on `ContentPresenter` currently. If using current master/nightly feed, I think it's now `TextElement.Foreground`.

**Comment by wnsntgs** [comment]
I tried style from https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/Button.xaml
```
  <Style Selector="Button:pointerover /template/ ContentPresenter#PART_ContentPresenter">
    <Setter Property="TextElement.Foreground" Value="Red" />
  </Style>
```
but I get error `[XAMLIL] Unable to resolve type TextElement from namespace https://github.com/avaloniaui`

It is a pity that the documentation is very superficial and poorly covers even basic things.

**Reply by timunie** [reply]
Hi @wnstngs 

You are right, that the docs are at some points quite lacking of some things. That said you get free support here and on telegram, gitter or discord. Please keep in mind that the team behind Avalonia is quite small and also is available for free. But you are free to contribute to the docs, if you want to. 

about your issue: Currently there are many things going on to prepare the next major release v11.0 the master branch shows already the chaged code. So TextElement will be available in v11.0. Currenty it is `TextBlock.Foreground`. 

https://github.com/AvaloniaUI/Avalonia/blob/582389e60b73559376f2378fe9dcd821d25991cc/src/Avalonia.Themes.Fluent/Controls/Button.xaml#L45-L49

Sorry for the inconvenience & happy coding
Tim
